race in automatic tests

Noticed while investigating why #25 is failing.

```
Script started on Fri Dec 13 17:18:26 2019
$ go test -timeout 120s -race -v ./cmd/...
=== RUN   TestMain
==================
WARNING: DATA RACE
Read at 0x00c0000e4090 by goroutine 9:
  github.com/neubot/dash/server.(*Handler).reaperLoop()
      /Users/sbasso/src/github.com/neubot/dash/server/server.go:382 +0x5e

Previous write at 0x00c0000e4090 by goroutine 8:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:25 +0x1fe
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

Goroutine 9 (running) created at:
  github.com/neubot/dash/server.(*Handler).StartReaper()
      /Users/sbasso/src/github.com/neubot/dash/server/server.go:396 +0x60
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:23 +0x17a
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

Goroutine 8 (running) created at:
  testing.(*T).Run()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:1202 +0xa6
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199
  testing.runTests()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:1200 +0x521
  testing.(*M).Run()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:1117 +0x2ff
  main.main()
      _testmain.go:44 +0x223
==================
==================
WARNING: DATA RACE
Write at 0x00000187f560 by goroutine 12:
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /Users/sbasso/go/pkg/mod/github.com/apex/log@v1.1.1/pkg.go:20 +0x899
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Previous read at 0x00000187f560 by goroutine 9:
  github.com/apex/log.(*Logger).log()
      /Users/sbasso/go/pkg/mod/github.com/apex/log@v1.1.1/logger.go:142 +0x42
  github.com/apex/log.(*Logger).Debug()
      /Users/sbasso/go/pkg/mod/github.com/apex/log@v1.1.1/entry.go:81 +0xd2
  github.com/neubot/dash/server.(*Handler).reaperLoop()
      /Users/sbasso/src/github.com/neubot/dash/server/server.go:382 +0x8d

Goroutine 12 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

Goroutine 9 (running) created at:
  github.com/neubot/dash/server.(*Handler).StartReaper()
      /Users/sbasso/src/github.com/neubot/dash/server/server.go:396 +0x60
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:23 +0x17a
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

==================

[...]

==================
WARNING: DATA RACE
Write at 0x00c00008a138 by goroutine 13:
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:965 +0x42
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Previous write at 0x00c00008a138 by goroutine 12:
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:965 +0x42
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Goroutine 13 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

Goroutine 12 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199
==================
==================
WARNING: DATA RACE
Write at 0x00c00008a150 by goroutine 13:
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:966 +0x5d
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Previous write at 0x00c00008a150 by goroutine 12:
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:966 +0x5d
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Goroutine 13 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

Goroutine 12 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199
==================
==================
WARNING: DATA RACE
Write at 0x00c000080e50 by goroutine 13:
  flag.(*stringValue).Set()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:231 +0x3a
  flag.(*FlagSet).parseOne()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:949 +0x9e8
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:968 +0x9a
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Previous write at 0x00c000080e50 by goroutine 12:
  flag.(*stringValue).Set()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:231 +0x3a
  flag.(*FlagSet).parseOne()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:949 +0x9e8
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:968 +0x9a
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Goroutine 13 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

Goroutine 12 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199
==================
==================
WARNING: DATA RACE
Write at 0x00c000082cc0 by goroutine 13:
  runtime.mapassign_faststr()
      /usr/local/Cellar/go/1.13.4/libexec/src/runtime/map_faststr.go:202 +0x0
  flag.(*FlagSet).parseOne()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:956 +0x59e
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:968 +0x9a
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Previous write at 0x00c000082cc0 by goroutine 12:
  runtime.mapassign_faststr()
      /usr/local/Cellar/go/1.13.4/libexec/src/runtime/map_faststr.go:202 +0x0
  flag.(*FlagSet).parseOne()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:956 +0x59e
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:968 +0x9a
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Goroutine 13 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

Goroutine 12 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199
==================
==================
WARNING: DATA RACE
Write at 0x00c0000859b8 by goroutine 13:
  flag.(*FlagSet).parseOne()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:956 +0x5b6
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:968 +0x9a
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Previous write at 0x00c0000859b8 by goroutine 12:
  flag.(*FlagSet).parseOne()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:956 +0x5b6
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:968 +0x9a
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Goroutine 13 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

Goroutine 12 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199
==================
==================
WARNING: DATA RACE
Write at 0x00c00008c2e8 by goroutine 13:
  flag.(*durationValue).Set()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:273 +0x79
  flag.(*FlagSet).parseOne()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:949 +0x9e8
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:968 +0x9a
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Previous write at 0x00c00008c2e8 by goroutine 12:
  flag.(*durationValue).Set()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:273 +0x79
  flag.(*FlagSet).parseOne()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:949 +0x9e8
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:968 +0x9a
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Goroutine 13 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

Goroutine 12 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199
==================
==================
WARNING: DATA RACE
Write at 0x00c00008c2c5 by goroutine 13:
  flag.(*boolValue).Set()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:121 +0x7f
  flag.(*FlagSet).parseOne()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:931 +0x512
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:968 +0x9a
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Previous write at 0x00c00008c2c5 by goroutine 12:
  flag.(*boolValue).Set()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:121 +0x7f
  flag.(*FlagSet).parseOne()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:931 +0x512
  flag.(*FlagSet).Parse()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:968 +0x9a
  github.com/neubot/dash/cmd/dash-client.internalmain()
      /usr/local/Cellar/go/1.13.4/libexec/src/flag/flag.go:996 +0x11c
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x48

Goroutine 13 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

Goroutine 12 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199
==================

[...]

==================
WARNING: DATA RACE
Write at 0x00c000080f30 by goroutine 13:
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x69

Previous write at 0x00c000080f30 by goroutine 12:
  github.com/neubot/dash/cmd/dash-client.TestMain.func1()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:38 +0x69

Goroutine 13 (running) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

Goroutine 12 (finished) created at:
  github.com/neubot/dash/cmd/dash-client.TestMain()
      /Users/sbasso/src/github.com/neubot/dash/cmd/dash-client/main_test.go:36 +0x47b
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199
==================

[...]

--- FAIL: TestMain (56.01s)
    testing.go:853: race detected during execution of test
FAIL
FAIL	github.com/neubot/dash/cmd/dash-client	56.126s
?   	github.com/neubot/dash/cmd/dash-server	[no test files]
FAIL
Script done on Fri Dec 13 17:22:14 2019
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,5 +8,5 @@ before_script:
 - go get github.com/mattn/goveralls
 
 script:
-- go test -v -coverprofile=coverage.cov -coverpkg=./... ./...
+- go test -race -v -coverprofile=coverage.cov -coverpkg=./... ./...
 - $GOPATH/bin/goveralls -coverprofile=coverage.cov -service=travis-ci

--- a/cmd/dash-client/main.go
+++ b/cmd/dash-client/main.go
@@ -45,15 +45,9 @@ var (
 	flagScheme = flag.String("scheme", "http", "Scheme to use")
 )
 
-func internalmain() error {
-	log.SetLevel(log.DebugLevel)
-	flag.Parse()
-	ctx, cancel := context.WithTimeout(context.Background(), *flagTimeout)
+func mainWithClientAndTimeout(client *client.Client, timeout time.Duration) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
-	client := client.New(clientName, clientVersion)
-	client.Logger = log.Log
-	client.FQDN = *flagHostname
-	client.Scheme = *flagScheme
 	ch, err := client.StartDownload(ctx)
 	if err != nil {
 		return err
@@ -74,6 +68,16 @@ func internalmain() error {
 	}
 	fmt.Printf("%s\n", string(data))
 	return nil
+}
+
+func internalmain() error {
+	log.SetLevel(log.DebugLevel)
+	flag.Parse()
+	client := client.New(clientName, clientVersion)
+	client.Logger = log.Log
+	client.FQDN = *flagHostname
+	client.Scheme = *flagScheme
+	return mainWithClientAndTimeout(client, *flagTimeout)
 }
 
 func main() {

--- a/cmd/dash-client/main_test.go
+++ b/cmd/dash-client/main_test.go
@@ -10,6 +10,7 @@ import (
 	"time"
 
 	"github.com/apex/log"
+	"github.com/neubot/dash/client"
 	"github.com/neubot/dash/server"
 )
 
@@ -20,22 +21,23 @@ func TestMain(t *testing.T) {
 	mux := http.NewServeMux()
 	handler := server.NewHandler("../../testdata")
 	ctx, cancel := context.WithCancel(context.Background())
+	handler.Logger = log.Log
 	handler.StartReaper(ctx)
 	handler.RegisterHandlers(mux)
-	handler.Logger = log.Log
 	server := httptest.NewServer(mux)
 	defer server.Close()
 	URL, err := url.Parse(server.URL)
 	if err != nil {
 		t.Fatal(err)
 	}
-	*flagHostname = URL.Host
 	var wg sync.WaitGroup
 	for i := 0; i < 17; i++ {
 		wg.Add(1)
 		go func(delay int) {
 			time.Sleep(time.Duration(delay) * time.Second)
-			err = internalmain()
+			client := client.New(clientName, clientVersion)
+			client.FQDN = URL.Host
+			err := mainWithClientAndTimeout(client, 55*time.Second)
 			if err != nil {
 				t.Fatal(err)
 			}

--- a/go.mod
+++ b/go.mod
@@ -9,10 +9,10 @@ require (
 	github.com/kr/pretty v0.1.0 // indirect
 	github.com/m-lab/go v1.1.0
 	github.com/m-lab/ndt7-client-go v0.0.0-20190724152841-ad7eefc52fe1
-	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
-	github.com/modern-go/reflect2 v1.0.1 // indirect
 	github.com/prometheus/client_golang v1.1.0 // indirect
 	github.com/stretchr/testify v1.4.0 // indirect
 	golang.org/x/crypto v0.0.0-20190426145343-a29dc8fdc734
 	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
 )
+
+go 1.13

